core/modules/comment/comment.theme.inc

```diff
@@ -109,6 +109,11 @@ function template_preprocess_comment(&$variables) {
   $variables['permalink'] = l(t('Permalink'), $uri['path'], $uri['options']);
   $variables['submitted'] = t('!datetime by !username', array('!datetime' => $variables['created'], '!username' => $variables['author']));
 
+  // Check to see if title should be hidden
+  if ($node_type->settings['comment_subject_hide']) {
+    $variables['title'] = '';
```

Comment:
Note that this will still create an empty `<h3></h3>` tag in the output. Not sure if that matters.

Reply:
Noted. Looking for ideas on how to avoid that.